contracts/poap-manager/src/contract.rs

```diff
@@ -0,0 +1,530 @@
+        let invalid_msg = InstantiateMsg {
+            admin: "a".into(),
+            poap_code_id: 1u64.into(),
+            poap_instantiate_msg: POAPInstantiateMsg {
+                admin: CREATOR.into(),
+                minter: CREATOR.into(),
+                cw721_code_id: 2u64.into(),
+                cw721_initiate_msg: Cw721InstantiateMsg {
+                    minter: CREATOR.into(),
+                    name: CREATOR.into(),
+                    symbol: CREATOR.into(),
+                },
+                event_info: EventInfo {
+                    creator: CREATOR.into(),
+                    start_time: Timestamp::from_seconds(10),
+                    end_time: Timestamp::from_seconds(20),
+                    per_address_limit: 2,
+                    base_poap_uri: "ipfs://popap-uri".to_string(),
+                    event_uri: "ipfs://event-uri".to_string(),
+                },
+            },
+        };
+        assert_eq!(
+            instantiate(deps.as_mut(), env, info, invalid_msg).unwrap_err(),
+            ContractError::Std(StdError::generic_err(
+                "Invalid input: human address too short"
+            ))
+        )
+    }
+
+    #[test]
+    fn instatiate_properly() {
+        let mut deps = mock_dependencies_with_custom_querier(&[]);
+        do_instantiate(deps.as_mut());
+
+        let config = CONFIG.load(&deps.storage).unwrap();
+        let expected = Config {
+            admin: Addr::unchecked("desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc"),
+            poap_code_id: 1u64,
+        };
+        assert_eq!(config, expected)
+    }
+
+    #[test]
+    fn poap_instantiate_with_invalid_reply_id_error() {
+        let mut deps = mock_dependencies_with_custom_querier(&[]);
+        do_instantiate(deps.as_mut());
+        let env = mock_env();
+        let result = reply(
+            deps.as_mut(),
+            env,
+            Reply {
+                id: 2,
+                result: SubMsgResult::Ok(SubMsgResponse {
+                    events: vec![],
+                    data: None,
+                }),
+            },
+        );
+        assert_eq!(result.unwrap_err(), ContractError::InvalidReplyID {},)
+    }
+
+    #[test]
+    fn poap_instantiate_with_invalid_instantiate_msg_error() {
+        let mut deps = mock_dependencies_with_custom_querier(&[]);
+        do_instantiate(deps.as_mut());
+        let env = mock_env();
+        let result = reply(
+            deps.as_mut(),
+            env,
+            Reply {
+                id: 1,
+                result: SubMsgResult::Ok(SubMsgResponse {
+                    events: vec![],
+                    data: None,
+                }),
+            },
+        );
+        assert_eq!(
+            result.unwrap_err(),
+            ContractError::ParseReplyError(ParseReplyError::ParseFailure(
+                "Missing reply data".into()
+            ))
+        )
+    }
+
+    #[test]
+    fn poap_instantiate_reply_properly() {
```

Comment:
This can be removed since already covered in the `instantiate_propery` integration test.